File: src/main/java/org/hildan/sudoku/solver/Assignment.java
package org.hildan.sudoku.solver;

import org.hildan.sudoku.model.Grid;

public class Assignment {
    
    public Grid grid;
    public int nbVisitedNodes;
    
    public Assignment(Grid grid) {
        super();
        this.grid = grid;
        this.nbVisitedNodes = 0;
    }
    
    public boolean isSolution() {
        return grid != null && grid.getEmptyTiles().isEmpty();
    }

    @Override
    public String toString() {
        if (isSolution())
            return grid.toString();
        return "No solution found.";
    }
}

File: src/main/kotlin/org/hildan/sudoku/solver/Assignment.kt
package org.hildan.sudoku.solver

import org.hildan.sudoku.drawing.format
import org.hildan.sudoku.model.Grid

class Assignment(
    val grid: Grid,
) {
    var nbVisitedNodes = 0
    val isSolution: Boolean
        get() = grid.isFull

    override fun toString(): String = if (isSolution) grid.format() else "No solution found."
}